core/pta/hwrng.c

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2018, Linaro Limited
+ * Copyright (c) 2021, EPAM Systems. All rights reserved.
+ *
+ * Based on plat-synquacer/rng_pta.c
+ *
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <rng_pta_client.h>
+#include <crypto/crypto.h>
+
+#define PTA_NAME "rng.pta"
+
+static TEE_Result rng_get_entropy(uint32_t types,
+				  TEE_Param params[TEE_NUM_PARAMS])
+{
+	uint8_t *e = NULL;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INOUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		DMSG("bad parameters types: 0x%" PRIx32, types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	e = (uint8_t *)params[0].memref.buffer;
+	if (!e)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	return crypto_rng_read(e, params[0].memref.size);
```

Comment:
NAK. This function is not suitable for this even if it under some circumstance might do the right thing.

Reply:
@jenswi-linaro 
Sorry, I think this rework from @sa-kib is linked to my previous comment and discussion around that function call. Still not clear to me what kind of circumstance you highlight here that may not do the right thing?

Reply:
`crypto_rng_read()` isn't guaranteed to provide raw hwrng entropy. If you need raw hwrng use `hw_get_random_byte()`.

Reply:
I'll drop two last fixups eddb770 & d9bd599 so that prev. use of `hw_get_random_byte()` restored.